app/src/main/java/com/amaze/filemanager/filesystem/files/FileUtils.java

```diff
@@ -416,6 +417,25 @@ public static float readableFileSizeFloat(long size) {
    */
   public static void installApk(
       final @NonNull File f, final @NonNull PermissionsActivity permissionsActivity) {
+
+    try {
+      PackageInstallValidation.validatePackageInstallability(f);
+    } catch (PackageInstallValidation.PackageCannotBeInstalledException e) {
+      Toast.makeText(
+              permissionsActivity,
+              R.string.error_google_play_cannot_update_myself,
+              Toast.LENGTH_LONG)
+          .show();
+      return;
+    } catch (IllegalStateException e) {
+      Toast.makeText(
```

Comment:
Question - in case we can't get package info, we still allow user to install the apk right?

Reply:
Yes, I think.